src/codeGenerator/exceptions.ts

```diff
@@ -0,0 +1,71 @@
+import { Expression } from "../ast/ast";
```

Comment:
Shouldn't these exceptions be in evaluator.ts?

Reply:
No strong opinion - evaluator.ts is getting fairly long so I thought organizing this into a new file might be nice

Reply:
Fair enough.